File: Desktop/exchange/src/main/java/com/kjb04/exchange/api/model/GroupMessage.java
package com.kjb04.exchange.api.model;

import com.google.gson.annotations.SerializedName;
import com.kjb04.exchange.DateParser;

import java.util.Date;

public class GroupMessage {
    @SerializedName("msg_id")
    private Integer msgId;
    @SerializedName("sender_username")
    private String senderUsername;
    @SerializedName("group_name")
    private String groupName;
    @SerializedName("content")
    private String content;
    @SerializedName("added_date")
    private String addedDate;


    public Integer getMsgId() {
        return msgId;
    }
    public String getSenderUsername() {
        return senderUsername;
    }
    public String getGroupName() {
        return groupName;
    }
    public String getContent() { return content; }
    public Date getAddedDate() {
        return DateParser.convertStringToDate(addedDate);
    }



    public GroupMessage(String content)
    {
        this.content = content;
    }
}

File: Android/exchange-android-EricNJ/app/src/main/java/com/example/currencyexchange/api/model/GroupMessage.kt
package com.example.currencyexchange.api.model

import com.google.gson.annotations.SerializedName
class GroupMessage {
    @SerializedName("msg_id")
    var msgId: Int? = null

    @SerializedName("sender_username")
    var senderUsername: String? = null

    @SerializedName("group_name")
    var groupName: String? = null

    @SerializedName("content")
    var content: String? = null

    @SerializedName("added_date")
    var addedDate: String? = null
}